### PR TITLE
Translate engineconn-plugins-shell  service classes from Scala to Java

### PR DESCRIPTION
**Pull request by ChengJie1053**
What is the purpose of the change
1, Reduce scala modules in engineconn-plugins-shell module.
2,Optimize and remove some code
3, Translate engineconn-plugins-shell	 service classes from Scala to Java using Chatgpt.

Related issues/PRs
Related issues: https://github.com/apache/linkis/issues/4192
Related pr:https://github.com/apache/linkis/issues/4192

Brief change log
1, Reduce scala modules in engineconn-plugins-shell	 module.
2, Translate engineconn-plugins-shell	 service classes from Scala to Java using Chatgpt.
3, finish test  engineconn-plugins-shell module

### CODE REVIEW

**Comment by peacewong** [comment]
ping ~ @jackxu2011 @CharlieYan24

File: linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ShellEngineConnExecutor.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    */
+    int pid = getPid(process);
+    GovernanceUtils.killProcess(String.valueOf(pid), "kill task " + taskID + " process", false);
+
+    /*
+     Kill yarn-applications
+    */
+    List<String> yarnAppIds = extractor.getExtractedYarnAppIds();
+    GovernanceUtils.killYarnJobApp(yarnAppIds);
+    logger.info("Finished kill yarn app ids in the engine of (" + getId() + ").}");
```
[the remark this answers is not in the document]
Comment (ChengJie1053) [reply linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ShellEngineConnExecutor.java:302]: Ok, thank you for reviewing the code

**Comment by peacewong** [comment]
After this PR is merged, is it possible to consider abstracting the public part of ShellEngineConnExecutor and ShellEngineConnConcurrentExecutor into an abstract class instead of inheriting the implementation.

File: linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ReaderThread.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    isReaderAlive = false;
+  }
+
+  @Override
+  public void run() {
+    String line = null;
+    List<String> logArray = new ArrayList<>();
+    while (true) {
+      try {
+        if (!((line = inputReader.readLine()) != null && isReaderAlive)) break;
```
[the remark this answers is not in the document]
Comment (ChengJie1053) [reply linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ReaderThread.java:71]: Thank you for reviewing the code

File: linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ShellEngineConnConcurrentExecutor.java [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+              wdStr);
+          workingDirectory = null;
+        }
+      } else {
+        workingDirectory = null;
+      }
+
+      String generatedCode;
+      if (argsArr == null || argsArr.length == 0) {
+        generatedCode = String.valueOf(generateRunCode(code));
```
[the remark this answers is not in the document]
Comment (ChengJie1053) [reply linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ShellEngineConnConcurrentExecutor.java:162]: Thank you ，I'll change the code

**Comment by ChengJie1053** [comment]
> After this PR is merged, is it possible to consider abstracting the public part of ShellEngineConnExecutor and ShellEngineConnConcurrentExecutor into an abstract class instead of inheriting the implementation.

I would like to do it

> After this PR is merged, is it possible to consider abstracting the public part of ShellEngineConnExecutor and ShellEngineConnConcurrentExecutor into an abstract class instead of inheriting the implementation.

Yes, I would like to do it

File: linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ShellEngineConnConcurrentExecutor.java [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+    return (etlHomeDir.exists() && etlHomeDir.isDirectory());
+  }
+
+  private String[] generateRunCode(String code) {
+    return new String[] {"sh", "-c", code};
+  }
+
+  private String[] generateRunCodeWithArgs(String code, String[] args) {
+    return new String[] {
+      "sh", "-c", "echo \"dummy " + StringUtils.join(args, " ") + "\" | xargs sh -c '" + code + "'"
```
[the remark this answers is not in the document]
Comment (ChengJie1053) [reply linkis-engineconn-plugins/shell/src/main/java/org/apache/linkis/manager/engineplugin/shell/executor/ShellEngineConnConcurrentExecutor.java:224]: Thank you ，I'll change the code

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**